src/device/wg_controller.rs

```diff
@@ -546,6 +550,42 @@ async fn build_requested_meshnet_peers_list<
     Ok(requested_peers)
 }
 
+/// Internal peers will never have IP collisions, but external peers can collide with both internal and external peers
+/// In case of collision, exclude the colliding IPs from external peers
+/// If a peer ends up not having any IPs after deduplicating, the peer will be unreachable
+fn deduplicate_peer_ips(peers: &[telio_model::config::Peer]) -> HashMap<PublicKey, Vec<IpAddr>> {
+    let mut peer_ips = HashMap::new();
+    let mut occupied_ips = vec![];
+    let mut external_peers = vec![];
+    let mut process_ips_for_peer = |peer: &telio_model::config::Peer| {
+        if let Some(ips) = &peer.ip_addresses {
+            let ips = ips
+                .iter()
+                .filter_map(|ip| {
+                    if occupied_ips.contains(ip) {
+                        None
```

Comment:
Could we add some `info` or `warn` level log message here if peer IP duplicate has been detected? This seems to be very useful information when analyzing logs in case of issues

Reply:
Maybe it would be nice to print some more detailed info about the collision? Also, if two peers would have the same sets of allowed IPs the current implementation drops the latter, shouldn't the algo try to resolve such situation? (Maybe not at any cost, not sure if the problem is easy.)